### PR TITLE
Fix and cleanup windows build (in Debug mode)

### PR DESCRIPTION
**Pull request by bellenot**
- `SetUpWindows.cmake`: Remove the `-D_WIN32` and `-D_WIN64` compiler flags that are set internally by the compiler and can cause redefinition warnings (e.g. with roottest)
 - `TWinNTSystem.cxx`: Don't add dependencies on system DLLs from the `C:\Windows` directory, their import libraries might not be available (e.g. in debug mode)

### CODE REVIEW

File: core/winnt/src/TWinNTSystem.cxx [hunk 1]
```diff
@@ -4319,10 +4319,12 @@ const char *TWinNTSystem::GetLibraries(const char *regexp, const char *options,
          // full path not found, so split it to extract the library name
          // only, set its extension to '.lib' and add it to the list of
          // libraries to search
```
Comment (Axel-Naumann) [inline core/winnt/src/TWinNTSystem.cxx:4321]: Can you update the comment to reflect the "path must not contain `C:\\WINDOWS`?

File: core/winnt/src/TWinNTSystem.cxx [hunk 2]
```diff
@@ -4319,10 +4319,12 @@ const char *TWinNTSystem::GetLibraries(const char *regexp, const char *options,
...
-         _splitpath(str.c_str(), drive, dir, fname, ext);
-         std::string libname(fname);
-         libname += ".lib";
-         all_libs.push_back(libname);
+         if (str.find("C:\\WINDOWS\\") == std::string::npos) {
```
Comment (Axel-Naumann) [inline core/winnt/src/TWinNTSystem.cxx:4322]: What if Windows installed on drive `D:`?

Comment (Axel-Naumann) [inline core/winnt/src/TWinNTSystem.cxx:4322]: Isn't there a Windows API to see what the Windows install folder is, or maybe even whether a DLL / path is a system one?

**Review by Axel-Naumann — APPROVED** [review]
But see comments!

File: core/winnt/src/TWinNTSystem.cxx (hunk 2 above)
Comment (bellenot) [reply core/winnt/src/TWinNTSystem.cxx:4322]: > What if Windows installed on drive `D:`?

Right, wrong patch (it should have been ` if (str.find(":\\WINDOWS\\")`

Comment (bellenot) [reply core/winnt/src/TWinNTSystem.cxx:4322]: Good point, thanks!

File: core/winnt/src/TWinNTSystem.cxx (hunk 1 above)
Comment (bellenot) [reply core/winnt/src/TWinNTSystem.cxx:4321]: Will do

**Pull request merged**